Modules/Quotas/test/Quotas.Domain.Tests/QuotaIdTests.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Backbone.Modules.Quotas.Domain.Aggregates.Identities;
+using FluentAssertions;
+using Xunit;
+
+namespace Backbone.Modules.Quotas.Domain.Tests;
+public class QuotaIdTests
+{
+    [Fact]
+    public void Can_create_quota_id_with_valid_value()
+    {
+        var validQuotaIdPrefix = "QUO";
+        var validIdLengthWithoutPrefix = 17;
+        var validIdValue = validQuotaIdPrefix + TestDataGenerator.GenerateString(validIdLengthWithoutPrefix);
+
+        var quotaId = QuotaId.Create(validIdValue);
+        var isQuotaIdValid = quotaId.IsSuccess;
+
+        isQuotaIdValid.Should().BeTrue();
+    }
+
+    [Fact]
+    public void Can_generate_quota_id()
+    {
+        var quotaId = QuotaId.Generate();
+        quotaId.Should().NotBeNull();
+    }
+
+    [Fact]
+    public void Cannot_create_quota_id_with_invalid_id_prefix()
+    {
+        var invalidQuotaIdPrefix = "QQQ";
+        var quotaId = QuotaId.Create(invalidQuotaIdPrefix + TestDataGenerator.GenerateString(17));
+
+        var isQuotaIdValid = quotaId.IsFailure;
+        var errorCode = quotaId.Error.Code;
+        var errorMessage = quotaId.Error.Message;
+
+        isQuotaIdValid.Should().BeTrue();
+        errorCode.Should().Be("error.platform.validation.invalidId");
+        errorMessage.Should().Contain("Id starts with");
+    }
+
+    [Fact]
+    public void Cannot_create_quota_id_with_invalid_id_length()
+    {
+        var validQuotaIdPrefix = "QUO";
+        var quotaIdValue = validQuotaIdPrefix + TestDataGenerator.GenerateString(QuotaId.DEFAULT_MAX_LENGTH);
+        var quotaId = QuotaId.Create(quotaIdValue);
+
+        var isQuotaIdValid = quotaId.IsFailure;
+        var errorCode = quotaId.Error.Code;
+        var errorMessage = quotaId.Error.Message;
+
+        isQuotaIdValid.Should().BeTrue();
+        errorCode.Should().Be("error.platform.validation.invalidId");
+        errorMessage.Should().Contain("Id has a length of");
+    }
+
+    [Fact]
+    public void Cannot_create_quota_id_with_invalid_id_characters()
+    {
+        char[] invalidCharacters = { '|', '-', '!' };
+        var validQuotaIdPrefix = "QUO";
+        var quotaIdValue = TestDataGenerator.GenerateString(QuotaId.DEFAULT_MAX_LENGTH_WITHOUT_PREFIX, invalidCharacters);
```

Comment:
I don't know if we had it like that in the already existing Id tests as well, but I don't like that this test is unpredictable in some cases.

Let's say we accidentally add "|" to the allowed characters of a QuotaId. When the randomly generated Id coincidentally generates an id that doesn't contain "|", the test will succeed. In all other cases, it will fail.

So you should rather create a `Theory`, which accepts an invalid character as a parameter, appends it to a string of valid characters, and tries to create a QuotaId with the result.
Define some invalid characters as `TestData`.